File: functions/src/models/slider.model.ts
export interface Slider {
  uid: string;
  judul: string;
  deskripsi: string;
  gambar: string;
  gambar_path: string;
  created_at: FirebaseFirestore.Timestamp;
}

export interface CreateSliderInput {
  judul: string;
  deskripsi: string;
  gambar_base64: string;
}

export interface DeleteSliderInput {
  uid: string;
}

File: module_user/lib/data/models/slider_model.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:module_core/shared_data/models/timestamp_converter.dart';
import '../../domain/entities/slider.dart';

part 'slider_model.g.dart';

@JsonSerializable()
@TimestampConverter()
class SliderModel extends SliderData {
  const SliderModel({
    required super.uid,
    required super.judul,
    required super.deskripsi,
    required super.gambar,
    @JsonKey(name: 'gambar_path') required super.gambarPath,
    @JsonKey(name: 'created_at') required super.createdAt,
  });

  factory SliderModel.fromJson(Map<String, dynamic> json) =>
      _$SliderModelFromJson(json);
  Map<String, dynamic> toJson() => _$SliderModelToJson(this);
}
